### PR TITLE
Milestone 7: Budget-App | Project Final Submission

### PR DESCRIPTION
**Pull request by samiullah997**
## Milestone 7: Budget-App | Project Final Submission
In this milestone I have :

- Installed Ruby on Rails framework.
- Understood Rails RESTful design and router.
- Used controllers to handle requests and render empty views.
- Used params from browser requests in a safe way.
- Used preprocessed HTML file with embedded Ruby code.
- Used layouts and templates for shared content.
- Used database migration files to maintain database schema.
- Used validations for models.
- Secured app from n+1 problems.
- Understood what ORM is.
- Written SQL queries with ActiveRecord.
- Set up associations between models
- Built a mobile web app that requires the user to log in.
- Used devise gem for authentication.
- Limited access to web app resources based on authorization rules.
- Analyzed in writing why you have made a coding choice using one structure over another.
- Implemented unit and integrations tests for main functionalities
- Future improvements
- Notifications that close for the alerts, errors, and notices
- Make a responsive side menu
## ⚠️ To Run our project
- Make sure to update the database.yml  file that will be used for database connection and migrations of development and test :
POSTGRES_USER=YourUsername
If you declared a password when creating the database:
POSTGRES_PASSWORD=YourPassword
POSTGRES_HOST=localhost
POSTGRES_DB=Budget_App_development
POSTGRES_TEST_DB=Budget_App_test
Happy reviewing 😄
Useful Links
Here is a link to the Zoom video [presentation](https://www.loom.com/share/7a069d3c9ba841f98177f4365011a177)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,138 +73,103 @@ After you're finished please remove all the comments and instructions!
...
+
+Here is a link to the [Video Link](https://www.loom.com/share/7a069d3c9ba841f98177f4365011a177) of the project.
 <!-- LIVE DEMO -->
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
-
-- [Live Demo Link](https://yourdeployedapplicationlink.com)
+- [Budget-App Coming soon...](https://budget-app.herokuapp.com/)
```
Comment (mikethreels) [inline README.md:89]: - [ ] when trying to open the live demo it is only displaying the word "Hi!" please make sure to deploy the project.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Samiullah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (samiullah997) [reply README.md:89]: Hi Mr. @mikethreels,
This link is not working I have to try to deploy this project but I am having some issues. for deployment the need to first add a payment method. when I add the details of my card then the card is declined and never saved. I will try to fix it later.

**Review by omarsalem7 — APPROVED** [review]
Hi @samiullah997 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ The live demo added
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by samiullah997** [comment]
Thanks, @omarsalem7 to give me good feedback. Happy coding 👋 .

**Pull request merged**